context/src/main/java/io/opentelemetry/context/Context.java

```diff
@@ -99,6 +99,23 @@ static Context root() {
     return ArrayBasedContext.root();
   }
 
+  /**
+   * Returns an {@link Executor} which delegates to the provided {@code executor}, wrapping all
+   * invocations with the {@linkplain Context#current() current context} at the time of invocation.
+   */
+  static Executor currentContextWrapping(Executor executor) {
```

Comment:
Naming is the hardest problem in eng, maybe consider `wrapCurrent` just in case we will add equivalent for wrapping with current Runnable/Callable and would be also consistent with current `wrap` helpers?

Reply:
> maybe consider wrapCurrent just in case we will add equivalent for wrapping with current Runnable/Callable and would be also consistent with current wrap helpers?

I don't think we'll be adding those since it's very easy to call `Context.current().wrap` but even if we did, I precisely don't want consistency with it :) This method uses the current context when calling ExecutorService.execute, not when calling this method so it's quite different.